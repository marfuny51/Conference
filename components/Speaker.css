.Table {
    text-align: center;
}

.TableSpeaker {
    border-collapse: collapse;
    width: 70em;
    height: 8em;
    margin: 2em auto 1em auto;
}

.TableSpeakerAdd {
    border-collapse: collapse;
    width: 70em;
    height: 8em;
    margin: 2em auto 1em auto;
    opacity:0.3;
}

.TableHead {
    color: #00D970;
}

.TableHead th {
    font-family: 'Muli', sans-serif;
    background-color: #127a16;
    color: white;
    letter-spacing:0.1em;
    padding: 0.6em;
    border: 0em;
}

.TableBody tr {
    background-color: #d3f1d4;
    border-bottom: 0.1em solid #127a16;
    padding: 0.1em;
    height: 3em;
}

.TableBody tr:hover {
    background-color: #f5f5f5;
}

.TableBody tr td {
    font-family: 'Muli', sans-serif;
    border: 0em;
}

.Read {
    font-family: 'Muli', sans-serif;
    height: 2.2em;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5em;
    border:0.2em solid #77b45b;
}

.Read:hover {
    font-weight: bold;
    height: 2.5em;
    border:0.2em solid #456835;
    cursor: pointer;
}

.Read2 {
    font-family: 'Muli', sans-serif;
    height: 2.2em;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5em;
    border:0.2em solid #77b45b;
    margin-right: 1.2em;
}

.Read2:hover {
    font-weight: bold;
    height: 2.5em;
    border:0.2em solid #456835;
    cursor: pointer;
}

.Add {
    font-family: 'Muli', sans-serif;
    font-size: 1em;
    font-weight: bold;
    height: 2.5em;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5em;
    border:0.2em solid #77b45b;
}

.Add:hover {
    font-size: 1.1em;
    border:0.2em solid #456835;
    cursor: pointer;
}

.NavLink {
    color: #172c11;
    font-weight: bold;
}

.LoadData {
    font-size: 1.1em;
    text-align: center;
    color: white;
}

.AddSpeaker {
    width: 45em;
    height: 13em;
    position: absolute;
    top: 35%;
    left: 32%;
    background-color: #d3f1d4;
    border: 0.2em solid #77b45b;
    border-radius: 0.5em;
}

.Title {
    font-family: 'Muli', sans-serif;
    font-weight: bold;
    letter-spacing:0.1em;
    padding-top: 1em;
}

.Input {
    font-family: 'Muli', sans-serif;
    text-align: left;
    padding: 1em 1em 1em 4em;
}

.Valide {
    color: red;
}

.Textarea {
    width: 20em;
    height: 7em;
}