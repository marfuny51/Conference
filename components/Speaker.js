import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import './Speakers.css';

class Speaker extends React.PureComponent {

  static propTypes = {
    speakers:PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.number.isRequired,
          name: PropTypes.string.isRequired,
          phone: PropTypes.string.isRequired,
          position: PropTypes.string.isRequired,
          topic: PropTypes.string.isRequired,
        })
      ),
  };
  
  render() {

    return (
    <tr key={this.props.info.id}>
        <td><NavLink to={"/speaker/"+this.props.info.name}>{this.props.info.name}</NavLink></td>
        <td>{this.props.info.phone}</td>
        <td>{this.props.info.position}</td>
        <td>{this.props.info.topic}</td>
    </tr>
    );

  }

}

export default Speaker;
