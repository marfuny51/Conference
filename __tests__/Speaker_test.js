import React from 'react';
import Speaker from '../components/Speaker';
import renderer from 'react-test-renderer';

test('работа Speaker', () => {

    let speaker={id:460,name:"Inokentii Zhivago",phone:"+375442569847",position:"Business analyst",topic:"Business analysis in software development"};
    // создаём тестовую версию компонента
    const component = renderer.create(
      <Speaker 
      speaker={speaker}
    />
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Edit'
    const buttonEdit = component.root.findByProps( {value:'Edit'}); 
    // и "нажмём" на неё
    buttonEdit.props.onClick();
    window.confirm = () => {true};
 
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Delete'
    const buttonDelete = component.root.findByProps( {value:'Delete'}); 
    // и "нажмём" на неё
    buttonDelete.props.onClick();
  
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    
    buttonEdit.props.onClick(); 
    
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    
    buttonDelete.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    /*
    // можно эмулировать события, передавая в качестве объекта события то что нам нужно:
    wrapper.find('select').simulate('change', {
      target: { value: "hello" },
    });
    */
  
  });
  