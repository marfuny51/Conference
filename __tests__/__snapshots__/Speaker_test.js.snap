// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Speaker 1`] = `
<tr>
  <td>
    Inokentii Zhivago
  </td>
  <td>
    +375442569847
  </td>
  <td>
    Business analyst
  </td>
  <td>
    Business analysis in software development
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Speaker 2`] = `
<tr>
  <td>
    Inokentii Zhivago
  </td>
  <td>
    +375442569847
  </td>
  <td>
    Business analyst
  </td>
  <td>
    Business analysis in software development
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Speaker 3`] = `
<tr>
  <td>
    Inokentii Zhivago
  </td>
  <td>
    +375442569847
  </td>
  <td>
    Business analyst
  </td>
  <td>
    Business analysis in software development
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Speaker 4`] = `
<tr>
  <td>
    Inokentii Zhivago
  </td>
  <td>
    +375442569847
  </td>
  <td>
    Business analyst
  </td>
  <td>
    Business analysis in software development
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Speaker 5`] = `
<tr>
  <td>
    Inokentii Zhivago
  </td>
  <td>
    +375442569847
  </td>
  <td>
    Business analyst
  </td>
  <td>
    Business analysis in software development
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;
